packages/katello/katello/hostname-change.rb

```diff
@@ -220,12 +220,15 @@ def using_custom_certs?(scenario_answers)
       scenario_answers["certs"]["server_key"]
     end
 
-    def delete_puppet_certs
+    def regenerate_puppet_certs
       return unless @scenario_answers['puppet'].is_a?(Hash)
 
       puppet_ssldir = @scenario_answers['puppet']['ssldir']
 
-      run_cmd("rm -rf '#{puppet_ssldir}'")
+      return unless puppet_ssldir
+
+      run_cmd("rm -rf '#{puppet_ssldir}'/*/#{@old_hostname}.*")
+      run_cmd("puppetserver ca generate --ca-client --certname #{@new_hostname}")
```

Comment:
Do we need to take any subject alt names into account here?

Reply:
We didn't in the past, we don't now ;)